src/lib/storage/reference_segment.cpp

```diff
@@ -0,0 +1,24 @@
+#include "reference_segment.hpp"
+
+namespace opossum {
+
+ReferenceSegment::ReferenceSegment(const std::shared_ptr<const Table>& referenced_table,
+                                   const ColumnID referenced_column_id, const std::shared_ptr<const PosList>& pos)
+    : _referenced_table(referenced_table), _referenced_column_id(referenced_column_id), _position_list(pos) {}
+
+ChunkOffset ReferenceSegment::size() const { return _position_list->size(); }
+
+size_t ReferenceSegment::estimate_memory_usage() const { return sizeof(RowID) * size(); }
```

Comment:
We weren't quite sure what to put in here, but we argued the following:

The ReferenceSegment does not own either the table or the position list, but only references them through pointers. That means we would not assign the size of either to the segment, as they could also be referenced by other segments. Therefore, our estimated memory usage comes down to the size of two shared pointers plus the size of a ColumnID instance.

Regardless of that, we believe that in your implementation it would be better to use ``capacity()`` of the PosList instead of ``size()`` and you are definitely missing/not accounting for the other two instance variables and their associated memory usage.